File: SDKs/googleapis.com/analyticsadmin/v1beta/typescript/src/sdk/models/shared/googleanalyticsadminv1betachangehistorychangechangehistoryresource.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleAnalyticsAdminV1betaAccount } from "./googleanalyticsadminv1betaaccount";
import { GoogleAnalyticsAdminV1betaConversionEvent } from "./googleanalyticsadminv1betaconversionevent";
import { GoogleAnalyticsAdminV1betaDataRetentionSettings } from "./googleanalyticsadminv1betadataretentionsettings";
import { GoogleAnalyticsAdminV1betaDataStream } from "./googleanalyticsadminv1betadatastream";
import { GoogleAnalyticsAdminV1betaFirebaseLink } from "./googleanalyticsadminv1betafirebaselink";
import { GoogleAnalyticsAdminV1betaGoogleAdsLink } from "./googleanalyticsadminv1betagoogleadslink";
import { GoogleAnalyticsAdminV1betaMeasurementProtocolSecret } from "./googleanalyticsadminv1betameasurementprotocolsecret";
import { GoogleAnalyticsAdminV1betaProperty } from "./googleanalyticsadminv1betaproperty";



// GoogleAnalyticsAdminV1betaChangeHistoryChangeChangeHistoryResource
/** 
 * A snapshot of a resource as before or after the result of a change in change history.
**/
export class GoogleAnalyticsAdminV1betaChangeHistoryChangeChangeHistoryResource extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=account" })
  account?: GoogleAnalyticsAdminV1betaAccount;

  @SpeakeasyMetadata({ data: "json, name=conversionEvent" })
  conversionEvent?: GoogleAnalyticsAdminV1betaConversionEvent;

  @SpeakeasyMetadata({ data: "json, name=dataRetentionSettings" })
  dataRetentionSettings?: GoogleAnalyticsAdminV1betaDataRetentionSettings;

  @SpeakeasyMetadata({ data: "json, name=dataStream" })
  dataStream?: GoogleAnalyticsAdminV1betaDataStream;

  @SpeakeasyMetadata({ data: "json, name=firebaseLink" })
  firebaseLink?: GoogleAnalyticsAdminV1betaFirebaseLink;

  @SpeakeasyMetadata({ data: "json, name=googleAdsLink" })
  googleAdsLink?: GoogleAnalyticsAdminV1betaGoogleAdsLink;

  @SpeakeasyMetadata({ data: "json, name=measurementProtocolSecret" })
  measurementProtocolSecret?: GoogleAnalyticsAdminV1betaMeasurementProtocolSecret;

  @SpeakeasyMetadata({ data: "json, name=property" })
  property?: GoogleAnalyticsAdminV1betaProperty;
}

File: SDKs/googleapis.com/analyticsadmin/v1beta/python/src/sdk/models/shared/googleanalyticsadminv1betachangehistorychangechangehistoryresource.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googleanalyticsadminv1betaaccount as shared_googleanalyticsadminv1betaaccount
from ..shared import googleanalyticsadminv1betaconversionevent as shared_googleanalyticsadminv1betaconversionevent
from ..shared import googleanalyticsadminv1betadataretentionsettings as shared_googleanalyticsadminv1betadataretentionsettings
from ..shared import googleanalyticsadminv1betadatastream as shared_googleanalyticsadminv1betadatastream
from ..shared import googleanalyticsadminv1betafirebaselink as shared_googleanalyticsadminv1betafirebaselink
from ..shared import googleanalyticsadminv1betagoogleadslink as shared_googleanalyticsadminv1betagoogleadslink
from ..shared import googleanalyticsadminv1betameasurementprotocolsecret as shared_googleanalyticsadminv1betameasurementprotocolsecret
from ..shared import googleanalyticsadminv1betaproperty as shared_googleanalyticsadminv1betaproperty


@dataclass_json
@dataclasses.dataclass
class GoogleAnalyticsAdminV1betaChangeHistoryChangeChangeHistoryResource:
    r"""GoogleAnalyticsAdminV1betaChangeHistoryChangeChangeHistoryResource
    A snapshot of a resource as before or after the result of a change in change history.
    """
    
    account: Optional[shared_googleanalyticsadminv1betaaccount.GoogleAnalyticsAdminV1betaAccount] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('account') }})
    conversion_event: Optional[shared_googleanalyticsadminv1betaconversionevent.GoogleAnalyticsAdminV1betaConversionEvent] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('conversionEvent') }})
    data_retention_settings: Optional[shared_googleanalyticsadminv1betadataretentionsettings.GoogleAnalyticsAdminV1betaDataRetentionSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataRetentionSettings') }})
    data_stream: Optional[shared_googleanalyticsadminv1betadatastream.GoogleAnalyticsAdminV1betaDataStream] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('dataStream') }})
    firebase_link: Optional[shared_googleanalyticsadminv1betafirebaselink.GoogleAnalyticsAdminV1betaFirebaseLink] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('firebaseLink') }})
    google_ads_link: Optional[shared_googleanalyticsadminv1betagoogleadslink.GoogleAnalyticsAdminV1betaGoogleAdsLink] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('googleAdsLink') }})
    measurement_protocol_secret: Optional[shared_googleanalyticsadminv1betameasurementprotocolsecret.GoogleAnalyticsAdminV1betaMeasurementProtocolSecret] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('measurementProtocolSecret') }})
    property: Optional[shared_googleanalyticsadminv1betaproperty.GoogleAnalyticsAdminV1betaProperty] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('property') }})
    
